src/main/java/org/fenixedu/academic/domain/accounting/Event.java

```diff
@@ -653,7 +642,7 @@ public Money getTotalAmountToPay() {
     }
 
     public Money getOriginalAmountToPay() {
-        return new Money(getDebtInterestCalculator(getWhenOccured().plusSeconds(1)).getDebtAmount());
+        return new Money(getDebtInterestCalculator(new DateTime()).getDebtAmount());
```

Comment:
😮😮😮 beware 😮😮😮

Reply:
The getDebtAmount() originally only iterates the "DebtStream" and all "Debt" objects are always added to the calculator, irregardless of the instance in time being considered; in other words, as long as the sum of values of the inner DueDateAmountMap remains the same then the getDebtAmount() will always produce the same value for every instance in time.

Either way, we ran a script for every existing event to check, and of course the value always matched.